docs/decisions/0002-authorization-model-foundation.rst

```diff
@@ -0,0 +1,195 @@
+  - Adding **new permissions (verbs)** that build on top of existing ones.
+  - Defining **new models/resources** (e.g., "workspace", "assignment") and expressing their relations to existing ones (e.g., platform → organization → course).
+
+* Applications must keep calling the same consistent check (e.g., *can(subject, action, object)*), while the schema or policy evolves underneath.
+
+VIII. Auditability
+=================
+
+Make all decisions explainable
+------------------------------
+* Every decision must have a trace:
+
+  - Which policy was used.
+  - Which attributes were checked.
+  - The effect (allow/deny).
+
+* Logs must let admins ask: "Why was this action allowed or denied?"
+* Traces must capture runtime values so audits remain possible later.
+* Permission checks in code must be **explicit and self-documenting**, so developers and stakeholders can easily understand how authorization is asked for in the system.
+
+IX. Security
+============
+
+Protect policies and logs against tampering
+--------------------------------------------
+
+* The system must guarantee the integrity of authorization policies and decision logs.
+* Policies and logs should be stored or managed in a way that makes tampering detectable.
+
+Consequences
+************
+1. **Strong audit needs.** We must build a central log of all decisions, including attributes and matched policies.
+2. **Attribute management.** ABAC requires attributes to be available and normalized. We must also capture their values in logs.
+3. **Scoped RBAC transition.** Some parts may use RBAC first, but the chosen system must support full ABAC.
+4. **Readable policies.** Even if technical, policies must be presented in a way non-technical people can review.
+5. **Scope consistency.** The system must provide a consistent definition and handling of scopes and resource hierarchies across all services, so that policies and checks have the same meaning everywhere.
+6. **Performance impact.** Logging and attributes add overhead. We must design caching and retention strategies.
+7. **Migration work.** Old in-code checks must be replaced step by step with policies.
+8. **Querying system.** The authorization model must support query-style checks (e.g., "list all objects this user can edit") at least as well as the current bridgekeeper system, either by integration or by providing equivalent functionality.
+
+Rejected Alternatives
+*********************
+* **RBAC-only**: too limited for contextual decisions.
+* **ReBAC**: rejected because it adds complexity and we lack strong use cases today.
+  - While ReBAC solves inheritance and recursive relationships well, it introduces complexity and a different way of thinking about authorization.
+* **In-code checks**: not auditable or shareable.
+* **Custom-built engine**: unnecessary when proven frameworks exist.
+
+References
+**********
+WIP
+
+Glossary
+********
```

Comment:
[suggestion] Alphabetizing this list will make it easier to find definitions

Reply:
Done, thanks!